pyvrp/minimise_fleet.py

```diff
@@ -0,0 +1,87 @@
+from collections import Counter
+
+from pyvrp._pyvrp import ProblemData, VehicleType
+from pyvrp.solve import SolveParams, solve
+from pyvrp.stop import FirstFeasible, MultipleCriteria, StoppingCriterion
+
+
+def minimise_fleet(
+    data: ProblemData,
+    stop: StoppingCriterion,
+    seed: int = 0,
+    params: SolveParams = SolveParams(),
+) -> list[VehicleType]:
+    """
+    Attempts to reduce the number of vehicles needed to achieve a feasible
+    solution to the given problem instance, subject to a stopping criterion.
+
+    Parameters
+    ----------
+    data
+        Problem instance with a given vehicle composition.
+    stop
+        Stopping criterion that determines how much effort to spend on finding
+        smaller fleet compositions.
+    seed
+        Seed value to use for the random number stream. Default 0.
+    params
+        Solver parameters to use. If not provided, a default will be used.
+
+    Returns
+    -------
+    list[VehicleType]
+        The smallest fleet composition admitting a feasible solution to the
+        problem instances that could be found before the stopping criterion was
+        hit. The original fleet is returned if no feasible solution was found.
+    """
+    feas_fleet = data.vehicle_types()
+
+    while True:
+        # TODO which vehicle to take out?
+        fleet = feas_fleet
+        fleet = _vehicles({0: fleet[0].num_available - 1}, fleet)
+        data = data.replace(vehicle_types=fleet)
```

Comment:
Not sure. This is easy in the case of a single vehicle type, but not so in general.